### PR TITLE
JS Environment testing

### PR DESCRIPTION
**Pull request by jsantell**
For #70 and #35, we want identical VM environments. Some scaffolding to test that across VMs

### CODE REVIEW

**Comment by jsantell** [comment]
Updated to target to state of the interpreter as a baseline

File: rust/common-runtime/tests/js_environment.rs [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+use serde_json::json;
+use shared::start_runtime;
+
+#[tokio::test]
+#[common_tracing]
+async fn it_has_an_empty_import_meta() -> Result<()> {
+    const TEST_IMPORT_OBJECT: &str = r#"
+import { read, write } from "common:io/state@0.0.1";
+
+export const run = () => {
```
Comment (jsantell) [inline rust/common-runtime/tests/js_environment.rs:19]: Could move these to test fixtures. Could also have a single "environment" module with switch statement on input to probe the context to reduce module builds in tests as well

**Comment by cdata** [comment]
@jsantell can you elaborate on the intended testing regime / progress metric and how you think it should be incorporated into our automated checks?

**Review by cdata — APPROVED** [review]

File: rust/common-runtime/tests/js_environment.rs (hunk 1 above)
Comment (cdata) [reply rust/common-runtime/tests/js_environment.rs:19]: Yeah, let's move it to `common-test-fixtures`! We already have another `common:module` fixture in there.

**Comment by jsantell** [comment]
> @jsantell can you elaborate on the intended testing regime / progress metric and how you think it should be incorporated into our automated checks?

Goals here are to snapshot the interpreter JS context such that we can audit future updates to Boa (decide if we want to enable some new JS global), ensure we are removing any features we'd like (Atomics?), test any intrinsic/bootstrapped JS features (e.g. provide our own `TextDecoder`). As we explore a Boa-only environment (#36), this is for verifying that the JS context is what we think it is (rather than the initial goals of unifying JS contexts across JS runtimes)

**Review by cdata — APPROVED** [review]

**Pull request merged**